File: stepik-java-api/src/main/java/org/stepik/api/objects/AbstractObject.java
package org.stepik.api.objects;

/**
 * @author meanmail
 */
public class AbstractObject {
    private long id;

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        AbstractObject that = (AbstractObject) o;

        return id == that.id;
    }

    @Override
    public int hashCode() {
        return (int) (id ^ (id >>> 32));
    }
}

File: stepik-java-api/src/main/java/org/stepik/api/objects/AbstractObject.kt
package org.stepik.api.objects


abstract class AbstractObject {
    open var id: Long = 0

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as AbstractObject

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id.hashCode()
    }

}
